roles/repositories/tasks/main.yml

```diff
@@ -0,0 +1,74 @@
+---
+- name: 'Enable Red Hat Repositories'
+  theforeman.foreman.repository_set:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs | default(omit) }}"
+    organization: "{{ organization }}"
+    product: "{{ item.0.name }}"
+    name: "{{ item.1.name }}"
+    state: enabled
+    repositories:
+      - releasever: "{{ item.1.releasever | default(omit) }}"
+        basearch: "{{ item.1.basearch | default(omit) }}"
+  with_subelements:
+    - "{{ products | selectattr('repository_sets', 'defined') | list }}"
+    - repository_sets
+
+- name: 'Create Products'
+  theforeman.foreman.product:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs | default(omit) }}"
+    organization: "{{ organization }}"
+    name: "{{ item.name }}"
+    state: present
+  with_items:
+    - "{{ products | selectattr('repositories', 'defined') | list }}"
+
+- name: 'Create Repositories'
+  theforeman.foreman.repository:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs | default(omit) }}"
+    organization: "{{ organization }}"
+    name: "{{ item.1.name }}"
+    url: "{{ item.1.url }}"
+    product: "{{ item.0.name }}"
+    content_type: "{{ item.1.content_type }}"
+    deb_components: "{{ item.1.deb_components | default(omit) }}"
+    deb_architectures: "{{ item.1.deb_architectures | default(omit) }}"
+    deb_releases: "{{ item.1.deb_releases | default(omit) }}"
+    download_policy: "{{ item.1.download_policy | default('on demand') }}"
```

Comment:
Any reason, we are overriding katello's default here?

Reply:
No and indeed it fails for Debian repositories unless I use `default(omit)` so I have made the change.